src/Modules/Microsoft.Azure.Functions.PowerShellWorker/Microsoft.Azure.Functions.PowerShellWorker.psm1

```diff
@@ -67,29 +77,127 @@ function Get-OutputBinding {
+function Get-DataCollectingBehavior
+{
+    param($BindingInfo)
+
+    # binding info not available
+    if ($null -eq $BindingInfo)
+    {
+        return [DataCollectingBehavior]::Scalar
+    }
+
+    switch ($BindingInfo.Type)
+    {
+        "http" { return [DataCollectingBehavior]::Scalar }
+        "blob" { return [DataCollectingBehavior]::Scalar }
+
+        "sendGrid" { return [DataCollectingBehavior]::Scalar }
+        "onedrive" { return [DataCollectingBehavior]::Scalar }
+        "outlook"  { return [DataCollectingBehavior]::Scalar }
+        "notificationHub" { return [DataCollectingBehavior]::Scalar }
+
+        "excel"    { return [DataCollectingBehavior]::Collection }
+        "table"    { return [DataCollectingBehavior]::Collection }
+        "queue"    { return [DataCollectingBehavior]::Collection }
+        "eventHub" { return [DataCollectingBehavior]::Collection }
+        "documentDB"  { return [DataCollectingBehavior]::Collection }
+        "mobileTable" { return [DataCollectingBehavior]::Collection }
+        "serviceBus"  { return [DataCollectingBehavior]::Collection }
+        "signalR"   { return [DataCollectingBehavior]::Collection }
+        "twilioSms" { return [DataCollectingBehavior]::Collection }
+        "graphWebhookSubscription" { return [DataCollectingBehavior]::Collection }
+
+        # Be conservative on new output bindings
+        default { return [DataCollectingBehavior]::Scalar }
     }
 }
 
+<#
+.SYNOPSIS
+    Combine the new data with the existing data for a output binding with 'Collection' behavior.
+    Here is what this command do:
+    - when there is no existing data
+      - if the new data is considered enumerable by PowerShell,
+        then all its elements get added to a List<object>, and that list is returned.
+      - otherwise, the new data is returned intact.
+
+    - when there is existing data
+      - if the existing data is a scalar, then a List<object> is created and the existing data
+        is added to the list.
+      - otherwise, the existing data is already a List<object>
+      - Then, depending on whether the new data is enumerable or not, its elements or itself will also be added to the list.
+      - That list is returned.
+#>
+function Merge-Collection
+{
+    param($OldData, $NewData)
+
+    $isNewDataEnumerable = [LanguagePrimitives]::IsObjectEnumerable($NewData)
+
+    if ($null -eq $OldData -and -not $isNewDataEnumerable)
+    {
+        return $NewData
+    }
+
+    $list = $OldData -as [System.Collections.Generic.List[object]]
+    if ($null -eq $list)
+    {
+        $list = [System.Collections.Generic.List[object]]::new()
+        if ($null -ne $OldData)
+        {
+            $list.Add($OldData)
+        }
+    }
+
+    if ($isNewDataEnumerable)
+    {
+        foreach ($item in $NewData)
+        {
+            $list.Add($item)
+        }
+    }
+    else
+    {
+        $list.Add($NewData)
+    }
+
+    return ,$list
```

Comment:
The `,` assures we always return a list, right?

Reply:
Yes, when getting here, we want to return the `List<object>` object and use it as the value, so we can reuse the list for the next `Push-OutputBinding`.
Without the comma, the list will be unrolled and turned to an array, and that will force us to create another list for the next `Push-OutputBinding`.